facade/disabledNodeFacade.go

```diff
@@ -0,0 +1,245 @@
+package facade
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/api/address"
+	"github.com/ElrondNetwork/elrond-go/api/block"
+	"github.com/ElrondNetwork/elrond-go/api/hardfork"
+	"github.com/ElrondNetwork/elrond-go/api/node"
+	transactionApi "github.com/ElrondNetwork/elrond-go/api/transaction"
+	"github.com/ElrondNetwork/elrond-go/api/validator"
+	"github.com/ElrondNetwork/elrond-go/api/vmValues"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/statistics"
+	"github.com/ElrondNetwork/elrond-go/core/vmcommon"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/data/vm"
+	"github.com/ElrondNetwork/elrond-go/debug"
+	"github.com/ElrondNetwork/elrond-go/heartbeat/data"
+	"github.com/ElrondNetwork/elrond-go/node/external"
+	"github.com/ElrondNetwork/elrond-go/ntp"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+var _ = address.FacadeHandler(&disabledNodeFacade{})
+var _ = hardfork.FacadeHandler(&disabledNodeFacade{})
+var _ = node.FacadeHandler(&disabledNodeFacade{})
+var _ = transactionApi.FacadeHandler(&disabledNodeFacade{})
+var _ = validator.FacadeHandler(&disabledNodeFacade{})
+var _ = vmValues.FacadeHandler(&disabledNodeFacade{})
+
+var errNodeStarting = errors.New("Node is starting")
+var emptStr string = ""
+
+// nodeFacade represents a facade for grouping the functionality for the node
+type disabledNodeFacade struct {
+}
+
+// NewDisableNodeFacade creates a new Facade with a NodeWrapper
+func NewDisableNodeFacade() (*disabledNodeFacade, error) {
+	return &disabledNodeFacade{}, nil
+}
+
+// SetSyncer sets the current syncer
+func (nf *disabledNodeFacade) SetSyncer(_ ntp.SyncTimer) {
+}
+
+// SetTpsBenchmark sets the tps benchmark handler
+func (nf *disabledNodeFacade) SetTpsBenchmark(_ statistics.TPSBenchmark) {
+}
+
+// TpsBenchmark returns the tps benchmark handler
+func (nf *disabledNodeFacade) TpsBenchmark() statistics.TPSBenchmark {
+	return nil
+}
+
+// StartBackgroundServices starts all background services needed for the correct functionality of the node
+func (nf *disabledNodeFacade) StartBackgroundServices() {
+}
+
+// RestAPIServerDebugMode return true is debug mode for Rest API is enabled
+func (nf *disabledNodeFacade) RestAPIServerDebugMode() bool {
+	return false
+}
+
+// RestApiInterface returns the interface on which the rest API should start on, based on the config file provided.
+// The API will start on the DefaultRestInterface value unless a correct value is passed or
+//  the value is explicitly set to off, in which case it will not start at all
+func (nf *disabledNodeFacade) RestApiInterface() string {
```

Comment:
This is due for removal, maybe add a TODO for this

Reply:
done